debezium-api/src/main/java/io/debezium/engine/DebeziumEngine.java

```diff
@@ -121,6 +121,36 @@ default void taskStopped() {
          * Should be called when a batch of records is finished being processed.
          */
         void markBatchFinished() throws InterruptedException;
+
+        /**
+         * Marks a record with updated source offsets as processed.
+         *
+         * @param record the record to commit
+         * @param sourceOffsets the source offsets to update the record with
+         */
+        void markProcessed(R record, Offsets sourceOffsets) throws InterruptedException;
+
+        /**
+         * Builds a new instance of an object implementing the {@link Offsets} contract.
+         *
+         * @return the object implementing the {@link Offsets} contract
+         */
+        Offsets buildOffsets();
+    }
+
+    /**
+     * Contract that should be passed to {@link RecordCommitter#markProcessed(Object, Offsets)} for marking a record
+     * as processed with updated offsets.
+     */
+    public static interface Offsets {
```

Comment:
```suggestion
    public interface Offsets {
```

It doesn't make a difference here.

Reply:
Updated.